src/Integration/Binding/ProjectBindingOperation.Writer.cs

```diff
@@ -45,22 +46,22 @@ internal partial class ProjectBindingOperation
         /// <param name="ruleSetFileName">The rule set file name</param>
         /// <param name="solutionRuleSet\">Full path of the parent solution-level SonarQube rule set</param>
         /// <returns>Full file path of the file that we expect to write to</returns>
-        internal /*for testing purposes*/ string QueueWriteProjectLevelRuleSet(string projectFullPath, string ruleSetFileName, ConfigFileInformation solutionRuleSet, string currentRuleSetPath)
+        internal /*for testing purposes*/ string QueueWriteProjectLevelRuleSet(string projectFullPath, string ruleSetFileName, IBindingConfigFile solutionRuleSet, string currentRuleSetPath)
         {
             Debug.Assert(!string.IsNullOrWhiteSpace(projectFullPath));
             Debug.Assert(!string.IsNullOrWhiteSpace(ruleSetFileName));
             Debug.Assert(solutionRuleSet != null);
 
             // TODO: clean up to remove down-cast to check for RuleSet
-            solutionRuleSet.BindingConfigFile.TryGetRuleSet(out var dotNetRuleSet);
+            solutionRuleSet.TryGetRuleSet(out var dotNetRuleSet);
```

Comment:
Not part of this change, but the "TODO" above is now redundant and can be removed later - we are now deliberately expecting a language-specific interface at this point.